ql/src/java/org/apache/hadoop/hive/ql/metadata/Table.java

```diff
@@ -1261,4 +1275,19 @@ public boolean equalsWithIgnoreWriteId(Table tbl ) {
     return result;
   }
 
+  public long getAsOfVersion() {
+    return asOfVersion;
+  }
+
+  public void setAsOfVersion(long asOfVersion) {
```

Comment:
nit: not that I want to complicate our own life, but shall we consider making the version field a string to be more generic, in case other storage handlers would implement this in the future, e.g. with git-like hash versions? Then we could accept any string and simply convert it to a long, if it's an Iceberg query. If we set in stone now that the version must be a number on the syntax level, then it'd become a breaking change later to amend that. But it's just food for thought, again, don't want to complicate life either

Reply:
Let's think about this, because this is an interesting question. One consequence is that:
- Instead of this `SELECT * FROM customers FOR SYSTEM_VERSION AS OF 12345678`
- We have to use `SELECT * FROM customers FOR SYSTEM_VERSION AS OF '12345678'`

Notice that we have to use `'` (quotes) for the string